physicsnemo/experimental/models/dit/layers.py

```
@@ -34,7 +48,395 @@
+        """
+        pass
+
+
+class TimmSelfAttention(AttentionModuleBase):
+    """Self-attention module that performs self-attention using the timm library implementation.
+    Expects an input tensor of shape (Batch, Sequence_Length, Hidden_Size) and 
+    returns a tensor of the same shape.
+
+    Parameters
+    ----------
+    hidden_size: int
+        The embedding dimension.
+    num_heads: int
+        Number of attention heads.
+    attn_drop_rate: float
+        The dropout rate for the attention module.
+    proj_drop_rate: float
+        The dropout rate for the projection module.
+    **kwargs: Any
+        Additional keyword arguments for the timm attention module.
+    """
+    def __init__(self, hidden_size: int, num_heads: int, attn_drop_rate: float = 0.0, proj_drop_rate: float = 0.0, **kwargs: Any):
+        super().__init__()
+        self.attn_op = Attention(dim=hidden_size, num_heads=num_heads, attn_drop=attn_drop_rate, proj_drop=proj_drop_rate, qkv_bias=True, **kwargs)
+
+    def forward(self, x: torch.Tensor, attn_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
+        """Forward pass of the TimmSelfAttention module.
+
+        Parameters
+        ----------
+        x: torch.Tensor
+            Input tensor of shape (Batch, Sequence_Length, Hidden_Size).
+        attn_mask: Optional[torch.Tensor]
+            The attention mask to apply to the input tensor (passed to timm's Attention module).
+            If None, no mask will be applied. This is only supported for timm version 1.0.16 and higher.
+
+        Returns
+        -------
+        torch.Tensor
+            Output tensor of shape (Batch, Sequence_Length, Hidden_Size).
+        """
+        if attn_mask is not None and not timm_v1_0_16:
+            raise ValueError("attn_mask in TimmSelfAttention is only supported for timm version 1.0.16 and higher")
+        
+        if not timm_v1_0_16:
+            return self.attn_op(x)
```

Would be better to avoid `mask` being silently ignored in this case, I think that could have undesirable effects for the user